### DISCUSSION TITLE
Is anyone able to use 'export_clips_as_files' without error on macOS [Monterey, Intel]

### DISCUSSION THREAD

**Discussion by jrluppold** [iluvcapra/py-ptsl#40] [Q&A]
When I use the 'export_clips_as_files' on Windows it works as expected.  In my use case I first use the 'select_all_clips_on_track' on a specific track & then use 'export_clips_as_files' into a temp directory.  On Windows this works as expected.

On macOS I get a CommandError of type **PT_UnknownError** with the message
"Assertion in \"/Users/autobld/gitlabci/PT/release/2024/r1/ProTools/NewFileLibs/Sys/MacOS/Sys_FileLocMacOS.cpp\", line 129.\n\n(Use the Copy button to copy this message to the clipboard.)"

I've googled the error message and find bits of advice that suggest making sure various directories are writable, but the advice is for different use cases than this.  
It 'seems' like ProTools is trying to tell me, at least when using the SDK, that it does not have permission to write files to any of the directories I'm asking it to export files to. I've tried changing the export directory to other paths than temp that I've explicitly given read/write permissions to.
When I perform the equivalent command on macOS within ProTools I also don't have a problem.
I've tried various other attempts to ensure the python app executing the command has the right permissions with no luck.

**Comment by iluvcapra** [comment]
Hey sorry I never got to this. Can you give me an example of the code you're using? I've seen this error when the file path isn't absolute.

**Reply by jrluppold** [reply]
Your comment prompted me to take a look at this again.  A thread you commented on at https://duc.avid.com/ and your comment on the method mentioning '(A MacOS path to a folder, must end with a colon ":".)'  led to me to the fix.  It was simply that the path needs to end in a trailing slash.  This isn't true for Windows. I'm a bit sad I initially overlooked your 'must end with colon' comment.

**Reply by iluvcapra** [reply]
That's a good catch but it's weird this is a requirement, they should fix that!
